fury/ui.py

```diff
@@ -5240,3 +5240,387 @@ def _set_position(self, coords):
         pass
         # self.actor.SetPosition(*coords)
         # self.container.SetPosition(*coords)
+
+
+class BulletList2D(UI):
+    """Display a list as bullet points
+    Attributes
+    ----------
+    _bullet_point: :class: 'BulletPoint2D'
+        Create a bullet point from node.
+    nodes: :class: 'List'
+        Store the generated bullet points in list
+    dict: :class: 'Dict'
+        Create a dictinory from nodes list.
+    """
+
+    def __init__(self, points=[], indent=10, draggable=False,
+                 position=(0, 0), size=(400, 500), bg_color=(0.5, 0.5, 0.5),
+                 bg_opacity=1, text_color=(0, 0, 0),
+                 bullet_bg_color=(1, 1, 1)):
+        """
+        Parameters
+        ----------
+        points: list
+            List of nodes that are to be displayed as BulletPoints
+        indent: int
+            Base indentation for the root node
+        draggable: Bool
+            If the card should be draggable
+        position : (float, float)
+            Absolute coordinates (x, y) of the lower-left corner of the
+            UI component
+        size : (int, int)
+            Width and height of the pixels of this UI component.
+        bg_color: (float, float, float)
+            Background color of card
+        bg_opacity: float
+            Background opacity
+        """
+
+        self.points = points
```

Comment:
You need to be extra careful here.
While assigning a list directly you are actually passing the reference and not the content, so any external changes to the original list will be reflected in your list as well.

I tried manipulating the original list externally and it seems your UI is unable to either throw an exception or update the UI.

Reply:
I'll pass the content of the original list instead of its reference, that's not a problem but updating the UI if original list is changed is a bit complicated. This is because how the UI element works. The UI has two lists, `self.points` (this list contains all the elements of the original list as is), `self.nodes` (this is a flat list that contains all the elements of the original list converted to `BulletPoint2D`). So, if I try to manipulate the original list this will not impact the UI as the UI element depends on `self.nodes` and not `self.points`. 
The only way of updating the UI when original list changes that I can think of is:-

1. check if original list and `self.points` are unequal right before adding the UI element to the scene
2. if they are unequal update `self.points` as well as `self.nodes`

Does this sound right?

Reply:
@Nibba2018 I think I need to add a custom event listener that listens for changes in any specific property in the UI element, as I couldn't find any information about this in vtk docs/examples. Do you think this is viable, if so where should it go, inside the interactor module or inside the base UI class as a `static` method?